app/controllers/repositories_controller.rb

```diff
@@ -0,0 +1,74 @@
+class RepositoriesController < ApplicationController
+  after_action :mark_everything_as_read, only: [:index, :show]
+  before_action :print_read_status, only: :index
+
+  @client = Octokit::Client.new(:access_token => ENV['PERSONAL_ACCESS_TOKEN'])
+
+  def index
+    # update_repo_tags
+    @repositories = Repository.all.sort_by(&:latest_release_date).reverse!
+    new
+    @repositories
+  end
+
+  def show
+    @repository = Repository.find(params[:id])
+    Services::Monitoring.set_repo_latest_release(@repository, @client)
+    @repository.save
+  end
+
+  def new
+    @repository = Repository.new
+  end
+
+  def create
+    puts "hello!"
+    @repository = Repository.new(repository_params)
+
+    if @repository.save
+      update_repo_tags
+      redirect_to repositories_path
+    else
+      render :new, status: :unprocessable_entity
+    end
+  end
+
+  def destroy
+    @repository = Repository.find(params[:id])
+    @repository.destroy
+
+    redirect_to root_path, status: :see_other
+  end
+
+  def update_repo_tags
```

Comment:
This method looks like it will be expensive as it iterates over all repositories, and it performs IO for each. Typically a controller action should be fast as otherwise it's the requester that ends up having to wait. I'd suggest trying to refactor so that any updates to the repositories is done asynchronously (and if an update needs to be shown to a user, then it can be done via some JavaScript foo/channels that let's the page know there is new content.